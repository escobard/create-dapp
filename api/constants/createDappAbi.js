const abi = [
  {
    "constant": true,
    "inputs": [
      {
        "name": "",
        "type": "uint256"
      }
    ],
    "name": "Payments",
    "outputs": [
      {
        "name": "user",
        "type": "address"
      },
      {
        "name": "amount",
        "type": "uint256"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [],
    "name": "paymentID",
    "outputs": [
      {
        "name": "",
        "type": "uint256"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "inputs": [],
    "payable": false,
    "stateMutability": "nonpayable",
    "type": "constructor"
  },
  {
    "constant": false,
    "inputs": [],
    "name": "makePayment",
    "outputs": [],
    "payable": true,
    "stateMutability": "payable",
    "type": "function"
  },
  {
    "constant": true,
    "inputs": [
      {
        "name": "_paymentID",
        "type": "uint256"
      }
    ],
    "name": "fetchPayment",
    "outputs": [
      {
        "name": "user",
        "type": "address"
      },
      {
        "name": "amount",
        "type": "uint256"
      }
    ],
    "payable": false,
    "stateMutability": "view",
    "type": "function"
  },
  {
    "constant": false,
    "inputs": [],
    "name": "emptyBalance",
    "outputs": [],
    "payable": true,
    "stateMutability": "payable",
    "type": "function"
  }
];

module.exports = abi;