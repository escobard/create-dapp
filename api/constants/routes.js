const routes = {
    port: process.env.PORT || 4000,
    health: '/health',
    postForm: '/postForm',
};

module.exports = routes;
