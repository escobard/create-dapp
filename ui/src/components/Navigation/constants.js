
export const items = [{ key: "home", active: true, name: "create dapp" }];

