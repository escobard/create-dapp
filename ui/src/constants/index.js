export const postFormFields = [
  {
    name: "stringType",
    label: "String Type",
    placeholder: "Enter a random string",
    value: "",
    error: false
  },
  {
    name: "stringLength",
    label: "String Length",
    placeholder: "Enter a random string, must have char length greater than 10",
    value: "",
    error: false
  },
  {
    name: "numberType",
    label: "Number Type",
    placeholder: "Enter a random number, must have char length greater than 10",
    value: "",
    error: false
  },
  {
    name: "numberMax",
    label: "Number Max",
    placeholder: "Enter a random number, must be greater than 10",
    value: "",
    error: false
  }
];

const environment =
  process.env.NODE_ENV === "production" ? "heroku" : "development";

console.log("environment", environment);

// TODO - to be updated with new heroku env for digipdfs
const apiRoot =
  environment === "heroku"
    ? "https://share-controller.herokuapp.com"
    : "http://localhost:4000";

console.log("root", apiRoot);

export const apiRoutes = {
  postForm: `${apiRoot + "/postForm"}`
};

export const headers = { "Access-Control-Allow-Origin": "*" };
