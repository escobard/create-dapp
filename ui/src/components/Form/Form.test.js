import React from "react";
import { shallow } from "enzyme";
import DynamicForm from "./index";
import { postFormFields } from "../../constants";

describe("Form snapshot renders", () => {
  it("should render makeDonation form and fields correctly", () => {
    const component = shallow(<DynamicForm fields={postFormFields} />);

    expect(component).toMatchSnapshot();
  });
});
