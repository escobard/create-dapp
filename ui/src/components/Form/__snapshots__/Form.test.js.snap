// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot renders should render makeDonation form and fields correctly 1`] = `
<Fragment>
  <Form
    as="form"
  >
    <Message />
    <FormGroup
      widths="equal"
    >
      <FormInput
        as={[Function]}
        control={[Function]}
        key="0"
        label="String Type"
        name="stringType"
        onChange={[Function]}
        placeholder="Enter a random string"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="1"
        label="String Length"
        name="stringLength"
        onChange={[Function]}
        placeholder="Enter a random string, must have char length greater than 10"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="2"
        label="Number Type"
        name="numberType"
        onChange={[Function]}
        placeholder="Enter a random number, must have char length greater than 10"
      />
      <FormInput
        as={[Function]}
        control={[Function]}
        key="3"
        label="Number Max"
        name="numberMax"
        onChange={[Function]}
        placeholder="Enter a random number, must be greater than 10"
      />
    </FormGroup>
    <FormField
      control={[Function]}
      onClick={[Function]}
    >
      Submit
    </FormField>
  </Form>
</Fragment>
`;
